File: AVc/app/src/main/java/com/hfad/avc/Util/CircularTransformation.java
package com.lastaurus.automatic_congratulations.util;

import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Shader;

import com.squareup.picasso.Transformation;

public class CircularTransformation implements Transformation {

   public CircularTransformation() {
   }

   @Override
   public Bitmap transform(final Bitmap source) {
      final Paint paint = new Paint();
      paint.setAntiAlias(true);
      paint.setShader(new BitmapShader(source, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP));

      final Bitmap output = Bitmap.createBitmap(source.getWidth(), source.getHeight(), Bitmap.Config.ARGB_8888);
      final Canvas canvas = new Canvas(output);
      int radius = source.getWidth() / 2;
      canvas.drawCircle(radius, source.getHeight() / 2, radius, paint);

      if (source != output)
         source.recycle();

      return output;
   }

   @Override
   public String key() {
      return "circle";
   }
}

File: app/src/main/java/com/lastaurus/automatic_congratulations/Util/CircularTransformation.kt
package com.lastaurus.automatic_congratulations.Util

import android.graphics.Bitmap
import android.graphics.BitmapShader
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Shader
import com.squareup.picasso.Transformation

class CircularTransformation : Transformation {
   override fun transform(source: Bitmap): Bitmap {
      val paint = Paint()
      paint.isAntiAlias = true
      paint.shader = BitmapShader(source, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP)
      val output = Bitmap.createBitmap(source.width, source.height, Bitmap.Config.ARGB_8888)
      val canvas = Canvas(output)
      val radius = source.width / 2
      canvas.drawCircle(radius.toFloat(), (source.height / 2).toFloat(), radius.toFloat(), paint)
      if (source != output) source.recycle()
      return output
   }

   override fun key(): String {
      return "circle"
   }
}